odoo/addons/base/tests/test_ir_mail_server_smtpd.py

```diff
@@ -301,21 +318,135 @@ def test_authentication_login_matrix(self):
             (True, PASSWORD, None),
         ]
 
-        for encryption in ('none', 'starttls', 'starttls_strict', 'ssl', 'ssl_strict'):
-            mail_server.smtp_encryption = encryption
+        for encryption, check_cert in (('none', False), ('starttls', False), ('starttls', True), ('ssl', False), ('ssl', True)):
             for auth_required, password, error_pattern in matrix:
-                mail_server.smtp_user = password and self.user_demo.email
-                mail_server.smtp_pass = password
+                self.mail_server_write({
+                    'smtp_encryption': encryption,
+                    'smtp_user': password and self.user_demo.email,
+                    'smtp_pass': password,
+                    'smtp_ssl_check_certificate': check_cert,
+                })
                 with self.subTest(encryption=encryption,
                                   auth_required=auth_required,
                                   password=password):
                     with self.start_smtpd(encryption, ssl_context, auth_required):
                         if error_pattern:
                             with self.assertRaises(UserError) as capture:
-                                mail_server.test_smtp_connection()
+                                self.mail_server.test_smtp_connection()
                             self.assertRegex(capture.exception.args[0], error_pattern)
                         else:
-                            mail_server.test_smtp_connection()
+                            self.mail_server.test_smtp_connection()
+
+    @mute_logger('mail.log')
+    def test_download_remote_certificate(self):
+        """
+        Add a security exception for a mail server that is using an
+        invalid certificate.
+        """
+        if TestCliMailServerSMTPD in type(self).mro():
```

Comment:
Instead of mro() move the test to the corresponding Test class (at the bottom of the file).